core/src/pixelcolor/transparent_color.rs

```diff
@@ -0,0 +1,155 @@
+///         let luma = (self.alpha as u16 * self.gray.luma() as u16) +
+///                    (255 - self.alpha as u16) * other.luma() as u16;
+///         Gray8::new((luma / 255) as u8)
+///     }
+/// }
+///
+/// // Make conversion available for `DrawTargetExt::color_converted`
+/// impl From<Gray8> for TransparentGray {
+///     fn from(value: Gray8) -> Self {
+///         TransparentGray { alpha: 255, gray: value }
+///     }
+/// }
+///
+/// // Mock transparent handling display
+/// pub struct Display {}
+/// impl DrawTarget for Display {
+///     type Color = TransparentGray;
+///     type Error = core::convert::Infallible;
+///
+///     fn draw_iter<I>(&mut self, pixels: I) -> Result<(), core::convert::Infallible>
+///     where
+///         I: IntoIterator<Item = Pixel<Self::Color>>,
+///     {
+///         for Pixel(point, color) in pixels.into_iter() {
+///             // mock existing pixel as middle gray
+///             let current_color = Gray8::new(127);
+///             // blend the color with existing color
+///             let final_color = color.blend_over(current_color);
+///             // ... draw final color on display
+///         }
+///         Ok(())
+///     }
+/// }
+/// impl OriginDimensions for Display {
+///     fn size(&self) -> Size {
+///         Size::new(300, 300)
+///     }
+/// }
+///
+/// // The image data.
+/// const DATA: &[u8] = &[
+///     0x80, 0xFF, 0xFF, 0x80, //
+///     0xFF, 0x00, 0x00, 0xFF, //
+///     0xFF, 0x00, 0x00, 0xFF, //
+///     0x80, 0xFF, 0xFF, 0x80, //
+/// ];
+///
+/// // Create a 4x4 non-transparent image
+/// let raw_image = ImageRaw::<Gray8>::new(DATA, Size::new(4,4)).unwrap();
+/// let image = Image::new(&raw_image, Point::zero());
+///
+/// // Create a transparent handling display
+/// let mut display = Display{};
+///
+/// // Draw a non-transparent object on a transparent handling display
+/// // Conversion is automatically handled by the `From` trait above and the `DrawTargetExt` trait
+/// image.draw(&mut display.color_converted()).unwrap();
+/// ```
+
+/// Transparent color trait.
+///
+/// This trait is used to represent color that handle transparency level.
+/// They can blend with other colors, transparent or nor.
+///
+/// Transparent colors are a special kind of [`PixelColor`].
+pub trait TransparentColor<C: PixelColor=Self>: PixelColor {
+    /// Blend this color over another underlying color
+    fn blend_over(self, other: C) -> C;
+}
+
+/// Trait for transparent colors that expose their alpha channel
+pub trait AlphaColor {
+    ///Returns the alpha channel value.
+    fn alpha(&self) -> u8;
+
+    /// The maximum value of the alpha channel.
+    const MAX_A: u8;
+}
+
+/// Trait used to find a transparent color that hold a pixelcolor with a compatible alpha channel
+/// Example:
+/// ```
+/// use embedded_graphics_core::pixelcolor::{Rgb888,HasAlphaColor,AlphaColor};
+/// let color = Rgb888::new(0x0,0x80,0xFF);
+/// let transparent = color.into_alpha_color(0);
```

Comment:
I don't really like the name `into_alpha_color`. What do you think about `with_alpha`? While the `with_` prefix is more common in constructors, there are some examples in the `std` library that use it for other purposes.
```suggestion
/// let transparent = color.with_alpha(0);
```
And it's common to use the same name for the trait as for the single method it contains: `WithAlpha`.

Reply:
yes